openassessment/assessment/models/base.py

```diff
@@ -109,7 +109,7 @@ def content_hash_from_dict(rubric_dict):
         rubric_dict.pop("content_hash", None)
 
         canonical_form = json.dumps(rubric_dict, sort_keys=True)
-        return sha1(canonical_form).hexdigest()
+        return sha1(canonical_form.encode('utf-8')).hexdigest()
```

Comment:
Need this change for this ticket
https://openedx.atlassian.net/browse/BOM-415
We have few other places also in this repo where sha1 is calling without encode. I just updated this for time being.

Reply:
Do we have made sure that `canonical_form` doesn't contain any non ASCII characters?
For example:
`'å®ßisøƒ†'.encode("utf-8")` would fail on Python 2.

Reply:
The same question for other places where `.encode("utf-8")` is used.

Reply:
@noraiz-anwar 

Python's json module simply takes the safe route and escapes non-ascii characters. This is a valid way of representing such characters in json, and any conforming parser will resurrect the proper Unicode characters when parsing the string:
```
import json
json.dumps({'Crêpes': 5})
'{"Cr\\u00eapes": 5}'
json.loads('{"Cr\\u00eapes": 5}')
{'Crêpes': 5}
```
https://stackoverflow.com/questions/35582528/python-encoding-and-json-dumps

Reply:
@noraiz-anwar Its highly unlikely that we will have ASCII characters in canonical_form as that is the rubric data. But @awais786 to be on safe side you can add the check 
```
if not isinstance(canonical_form, bytes):
    canonical_form = canonical_form.encode('utf-8')
```
before the sha1 call.

Reply:
That is interesting. So, basically we need bytes for `.hexdigest()` and for that, we are doing `.encode("utf-8")`

Reply:
@noraiz-anwar yes.